apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,217 @@
+import { db } from "@/lib/db";
+import { env } from "@/lib/env";
+
+import type { Readable } from "node:stream";
+import { clerkClient } from "@clerk/nextjs";
+import { sha256 } from "@unkey/hash";
+import { Resend } from "@unkey/resend";
+import type { NextApiRequest, NextApiResponse } from "next";
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+  runtime: "nodejs",
+};
+const { RESEND_API_KEY, GITHUB_KEYS_URI } = env();
+
+if (!RESEND_API_KEY || !GITHUB_KEYS_URI) {
+  throw new Error("Missing required environment variables");
+}
+const resend = new Resend({
+  apiKey: RESEND_API_KEY,
+});
+
+type Key = {
+  key_identifier: string;
+  key: string;
+  is_current: boolean;
+};
+// Needs to be tested when Github is live
+const verifyGitSignature = async (payload: string, signature: string, keyId: string) => {
+  //Check payload
+  if (!payload || payload.length === 0) {
+    throw new Error("No payload found");
+  }
+  //Check signature
+  if (!signature || signature.length === 0 || typeof signature !== "string") {
+    throw new Error("No signature found");
+  }
+  //Check KeyID
+  if (!keyId || keyId.length === 0 || typeof keyId !== "string") {
+    throw new Error("No KeyID found");
+  }
+  //Get data from Github url
+  const gitHub = await fetch(GITHUB_KEYS_URI);
+  if (!gitHub.ok) {
+    throw new Error("No public keys found");
+  }
+  const gitBody = await gitHub.json();
+  //Get public keys from Github
+  const publicKey =
+    gitBody.public_keys.find((k: Key) => {
+      if (k.key_identifier === keyId) {
+        return k;
+      }
+    }) ?? null;
+
+  if (!publicKey) {
+    console.error("No public keys found");
+  }
+  // Verify signature
+  const verify = crypto.createVerify("SHA256").update(payload);
+  return !verify.verify(publicKey.key, Buffer.from(signature.toString(), "base64"));
+};
+
+export default async function handler(request: NextApiRequest, response: NextApiResponse) {
+  const signature = request.headers["github-public-key-signature"]?.toString();
+  const keyId = request.headers["github-public-key-identifier"]?.toString();
+  const rawBody = await getRawBody(request);
+  const data = JSON.parse(Buffer.from(rawBody).toString("utf8"));
+
+  if (!signature) {
+    throw new Error("No signature found");
+  }
+  if (!keyId) {
+    throw new Error("No KeyID found");
+  }
+  if (data?.length === 0) {
+    throw new Error("No data found");
+  }
+  const isGithubVerified = await verifyGitSignature(rawBody.toString(), signature, keyId);
+  if (!isGithubVerified) {
+    throw new Error("Github signature not verified");
+  }
+
+  for (const item of data) {
+    const token = item.token.toString();
+    const hashedToken = await sha256(token);
+    const isKeysFound = await db.query.keys.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(eq(table.hash, hashedToken), isNull(table.deletedAt)),
+    });
+    if (!isKeysFound) {
+      throw new Error("No keys found");
```

Comment:
that's not an error, it's going to happen a lot I assume.
If we throw an error and not return a proper response to github, will they retry? or what happens?